osbs/build/plugins_configuration.py

```diff
@@ -0,0 +1,583 @@
+"""
+Copyright (c) 2018 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from __future__ import print_function, absolute_import, unicode_literals
+
+import logging
+import os
+import json
+import copy
+
+from osbs.constants import (BUILD_TYPE_ORCHESTRATOR, ISOLATED_RELEASE_FORMAT)
+from osbs.exceptions import OsbsException, OsbsValidationException
+from osbs.utils import Labels
+
+logger = logging.getLogger(__name__)
+
+
+class PluginsConfiguration(object):
+    def __init__(self, user_params):
+        # Figure out inner template to use from user_params:
+        self.user_params = user_params
+
+        #    <build_type>_inner:<arrangement_version>.json
+        arrangement_version = self.user_params.arrangement_version.value
+        build_type = self.user_params.build_type.value
+        self._template_path = '{0}_inner:{1}.json'.format(build_type, arrangement_version)
+        self._template = None
+
+    @property
+    def template(self):
+        if self._template is None:
+            path = os.path.join(self.user_params.build_json_dir.value, self._template_path)
+            logger.debug("loading template from path %s", path)
+            try:
+                with open(path, "r") as fp:
+                    self._template = json.load(fp)
+            except (IOError, OSError) as ex:
+                raise OsbsException("Can't open template '%s': %s" %
+                                    (path, repr(ex)))
+        return self._template
+
+    def remove_plugin(self, plugin_type, plugin_name):
+        """
+        if config contains plugin, remove it
+        """
+        for p in self.template[plugin_type]:
+            if p.get('name') == plugin_name:
+                self.template[plugin_type].remove(p)
```

Comment:
Just an idea, but `remove_plugin`, `add_plugin` and all `template_*` methods could be grouped together in a new class to better isolate concerns.

Reply:
Fixed.